system/extensions/host/run-image-boot-sh.toit

```diff
@@ -92,8 +92,15 @@ for (( ; ; )); do
   # Run the image.
   export TOIT_FLASH_UUID_FILE=\$PREFIX/\$current/uuid
   export TOIT_FLASH_REGISTRY_FILE=\$PREFIX/flash-registry
-  (cd \$PREFIX/\$current; ./run-image \$PREFIX/\$current \$PREFIX/scratch)
+  pushd \$PREFIX/\$current
+  ./run-image \$PREFIX/\$current \$PREFIX/scratch &
+  popd
+  RUN_IMAGE_PID=\$!
+  # Make sure to kill the run-image process if we are killed.
+  trap "kill \$RUN_IMAGE_PID; exit" TERM
```

Comment:
Isn't there a race condition here if we get killed at an unfortunate moment?

Also, doesn't this mean that we will not kill the current process on TERM, because we just handle that instead?